### PR TITLE
Add `OPTForQuestionAnswering`

### PR DESCRIPTION
**Pull request by clementapa**
# What does this PR do?

This PR adds `OPTForQuestionAnswering` in Transformers. The implementation is based on `BloomForQuestionAnswering` #19310. This introduces a new autoregressive model for question answering tasks in the library.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger @LysandreJik @ArthurZucker @younesbelkada

### CODE REVIEW

File: src/transformers/models/opt/modeling_opt.py [hunk 1]
```diff
@@ -1109,3 +1114,110 @@ def get_input_embeddings(self):
...
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(OPT_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        processor_class=_TOKENIZER_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_SEQUENCE_CLASSIFICATION,
+        output_type=SequenceClassifierOutputWithPast,
+        config_class=_CONFIG_FOR_DOC,
+        expected_output=_SEQ_CLASS_EXPECTED_OUTPUT,
+        expected_loss=_SEQ_CLASS_EXPECTED_LOSS,
```
Comment (sgugger) [inline src/transformers/models/opt/modeling_opt.py:1144]: You will need to adapt this here to use:
```py
        qa_target_start_index=_QA_TARGET_START_INDEX,
        qa_target_end_index=_QA_TARGET_END_INDEX,
        expected_output=_QA_EXPECTED_OUTPUT,
        expected_loss=_QA_EXPECTED_LOSS,
```
(see for instance the BERT model) and define those four constants.

**Review by sgugger — COMMENTED** [review]
Thanks a lot for working on this new model! We just need to fix the docstring and it should be good to merge!

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Comment by sgugger** [comment]
Failures are unrelated to this PR, so merging :-)

**Comment by ashim95** [comment]
Hi @clementapa , 
While adding `OPTForQuestionAnswering` did you test if you were able to train (say, fine-tune on squad) a QA model for any of the opt variants? 

I am getting a fast tokenizer error here: https://github.com/huggingface/transformers/blob/main/examples/pytorch/question-answering/run_qa.py#L345 

Essentially, the `run_qa.py` script requires the model to have a fast tokenizer which is not available for the OPT models.  

Thanks,

**Comment by ArthurZucker** [comment]
Hey! The Fast tokenizer is available for OPT. Make sure you are using main, as a recent issue with automatic conversion for OPT tokenizer was fixed. See #20823

**Pull request merged**